src/java.base/share/conf/security/java.security

```diff
@@ -158,10 +158,34 @@ RestrictedSecurity.NSS.140-2.securerandom.algorithm = PKCS11
 #endif
 
 #if defined aix-ppc || defined linux-ppc || defined linux-s390 || defined linux-x86 || defined windows
+#
+# Strict RestrictedSecurity mode profile for FIPS 140-3. This policy represents only allowable
+# approved cryptography in the OpenJCEPlusFIPS provider along with other non-cryptographic algorithms
+# commonly used. Please note that the following constraints represent algorithms no longer allowable
+# within this strict profile for the OpenJCEPlusFIPS provider as compared to the
+# "RestrictedSecurity.OpenJCEPlusFIPS.FIPS140-3-Weakly-Enforced" profile defined below.
+#    {AlgorithmParameterGenerator, DiffieHellman, *}, \
+#    {AlgorithmParameters, DiffieHellman, *}, \
+#    {KeyAgreement, DiffieHellman, *}, \
+#    {KeyFactory, DiffieHellman, *}, \
+#    {KeyGenerator, HmacSHA3-224, *}, \
+#    {KeyGenerator, HmacSHA3-256, *}, \
+#    {KeyGenerator, HmacSHA3-384, *}, \
+#    {KeyGenerator, HmacSHA3-512, *}, \
+#    {KeyPairGenerator, DiffieHellman, *}, \
+#    {MessageDigest, MD5, *}, \
+#    {MessageDigest, SHA-1, *}, \
+#    {MessageDigest, SHA-512/224, *}, \
+#    {MessageDigest, SHA-512/256, *}, \
+#    {Signature, NONEwithDSA, *}, \
+#    {Signature, NONEwithRSA, *}, \
+#    {Signature, RSAforSSL, *}, \
+#    {Signature, SHA1withRSA, *}
+#
 RestrictedSecurity.OpenJCEPlusFIPS.FIPS140-3.desc.name = OpenJCEPlusFIPS Cryptographic Module FIPS 140-3
-RestrictedSecurity.OpenJCEPlusFIPS.FIPS140-3.desc.default = true
+RestrictedSecurity.OpenJCEPlusFIPS.FIPS140-3.desc.default = false
```

Comment:
Why do we want this change? If the goal is to arrange that there's only one profile that says `default = true`, then `NSS.140-2` needs to change as well (both `NSS.140-2` and `OpenJCEPlusFIPS.FIPS140-3-Weakly-Enforced` currently say `default = true`).

Reply:
Looking at `RestrictedSecurity` some more, it seems that if one uses `-Dsemeru.customprofile=XXX` where `XXX` does **_not_** contain `'.'`, then all profiles beginning with `XXX` are considered; we want exactly one of those with the same prefix (before  the first dot) to say `default = true`. Do I have that right?

I'm not convinced all that extra complexity is worthwhile, but it's where we are.

Reply:
I'd like a response on this topic before we proceed.

Reply:
Thank you for the responses.

I think it's clearer and there's less potential for surprises if we just expected users to supply full profile names (e.g. `-Dsemeru.customprofile=NSS.140-2` or `-Dsemeru.customprofile=OpenJCEPlusFIPS.FIPS140-3-Weakly-Enforced`). The extension mechanism we have now would allow for shorter names (that extend longer ones), but then users would always provide the full name (even if it's short) of the desired profile, however, that should be a separate discussion.